File: java/src/main/java/com/lidong/javaops/algorithm/util/ArrayUtil.java
package com.lidong.javaops.algorithm.util;

/**
 * @author Ls J
 * @version 2019/4/29 15:09
 */
public class ArrayUtil {
    /**
     * 数组打印
     *
     * @param arr
     */
    public static void printArray(int[] arr) {
        for (int a : arr) {
            System.out.print(a + " ");
        }
        System.out.println();
    }

    /**
     * 交换两个元素
     * @param arr 目标数组
     * @param first 第一个元素的位置
     * @param second 第二个元素的位置
     */
    public static void swap(int[] arr, int first, int second) {
        int tmp = arr[first];
        arr[first] = arr[second];
        arr[second] = tmp;
    }
}

File: algorithm-go/src/lib/util/array_util.go
//
// @author Ls J
// @date 2020-05-31 23:48
//
package util

import "fmt"

//
// 交换两个元素的值
//
func Swap(arr []int, a, b int) {
    arr[a], arr[b] = arr[b], arr[a]
}

//
// 顺序输出
//
func Show(arr []int) {
    for _, value := range arr {
        fmt.Printf("%d\t", value)
    }
}
